lib/model/QubitDigitalObject.php

```diff
@@ -1386,53 +1386,121 @@ public function writeToFileSystem($asset, $options = array())
    * Download external file via sfWebBrowser and return its temporary location
    *
    * @param string URI
-   * @return mixed FALSE if error, array otherwise
+   * @param array options optional arguments
+   *
+   * @return string contents
    */
-  private function downloadExternalObject($uri)
+  private function downloadExternalObject($uri, $options = array())
   {
-    // Parse URL into components and get file/base name
-    $uriComponents = parse_url($uri);
-
     // Initialize web browser
     $timeout = sfConfig::get("app_download_timeout");
     $browser = new sfWebBrowser(array(), null, array('Timeout' => $timeout));
```

Comment:
Should we explicitly pass the desired adapter to the constructor? e.g.

```php
$browser = new sfWebBrowser(array(), 'sfCurlAdapter`, array('Timeout' => $timeout));
```

In AtoM, `sfCurlAdapter` will probably be the adapter being used as `php-curl` is listed as a dependency and it's the preferred adapter of `sfWebBrowser`, but I think it'd be clearer to express that constraint here as you are relying on curl-specific features a few lines below?